Improve over(under)lay brushes

Aurelien30000 opened the pull request:
## Overview
<!--  Please describe which issue this pull request targets.

If there is no issue, delete the "Fixes" part.
-->

Fixes #280

## Description
<!-- Please describe what this pull request does. -->
- Use boolean array instead of int array to keep track of already found blocks.
- Add origin air checks (as suggested by @OneLiteFeather) for overlay brushes when all blocks parameter is being used.

```[tasklist]
### Submitter Checklist
- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).
```

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/OverlayBrush.java:
```diff
@@ -156,7 +156,7 @@ private void overlayTwo(Snipe snipe) {
                 int blockZ = targetBlock.getZ();
                 for (int y = blockY; y >= editSession.getMinY() && !surfaceFound; y--) { // start scanning from the height you clicked at
-                    if (memory[x + brushSize][z + brushSize] != 1) { // if haven't already found the surface in this column
+                    if (!memory[x + brushSize][z + brushSize]) { // if haven't already found the surface in this column
```
The `!` can be overread, maybe a variable helps here :D 
e.x:
```java
var noSurfaceFound = !memory[x + brushSize][z + brushSize];
if(noSurfaceFound) // ...
```

Aurelien30000 replied:
FAVS is full of overlapping code horrors :sweat_smile: But yeah, let's improve it step by step, thanks for the review.

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/OverlayBrush.java:
```diff
@@ -169,21 +169,24 @@ private void overlayTwo(Snipe snipe) {
-                                            ); // fills down as many layers as you specify in
-                                            // parameters
-                                            memory[x + brushSize][z + brushSize] = 1; // stop it from checking any other blocks in this vertical 1x1 column.
+                                        if (!Materials.isEmpty(type)) {
```
see above comment

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/UnderlayBrush.java:
```diff
@@ -98,35 +100,37 @@ private void underlay(Snipe snipe) {
                 int blockZ = targetBlock.getZ();
                 for (int y = blockY; y < blockY + this.depth; y++) { // start scanning from the height you clicked at
-                    if (memory[x + brushSize][z + brushSize] != 1) { // if haven't already found the surface in this column
+                    if (!memory[x + brushSize][z + brushSize]) { // if haven't already found the surface in this column
```
see above comment

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/UnderlayBrush.java:
```diff
@@ -150,33 +154,38 @@ private void underlay2(Snipe snipe) {
                 int blockZ = targetBlock.getZ();
                 for (int y = blockY; y < blockY + this.depth; y++) { // start scanning from the height you clicked at
-                    if (memory[x + brushSize][z + brushSize] != 1) { // if haven't already found the surface in this column
+                    if (!memory[x + brushSize][z + brushSize]) { // if haven't already found the surface in this column
```
see above comment

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/UnderlayBrush.java:
```diff
@@ -150,33 +154,38 @@ private void underlay2(Snipe snipe) {
-                                    ); // fills down as many layers as you specify in
-                                    // parameters
-                                    memory[x + brushSize][z + brushSize] = 1; // stop it from checking any other blocks in this vertical 1x1 column.
+                                if (!Materials.isEmpty(type)) {
```
see above comment

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/splatter/SplatterOverlayBrush.java:
```diff
@@ -209,7 +209,7 @@ private void splatterOverlay(Snipe snipe) {
                 for (int y = targetBlock.getY(); y >= editSession.getMinY(); y--) {
                     // start scanning from the height you clicked at
-                    if (memory[x + brushSize][z + brushSize] != 1) {
+                    if (!memory[x + brushSize][z + brushSize]) {
```
see above comment

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/splatter/SplatterOverlayBrush.java:
```diff
@@ -311,19 +311,17 @@ private void splatterOverlayTwo(Snipe snipe) {
-                                z,
-                                2
-                        )) <= brushSizeSquared && splat[x + brushSize][z + brushSize] == 1) { // if inside of the column...&& if to be splattered
+                    if (!memory[x + brushSize][z + brushSize]) { // if haven't already found the surface in this column
```
see above comment

TheMeinerLP commented on src/main/java/com/thevoxelbox/voxelsniper/brush/type/performer/splatter/SplatterOverlayBrush.java:
```diff
@@ -335,36 +333,37 @@ private void splatterOverlayTwo(Snipe snipe) {
-                                            ); // fills down as many layers as you specify in
-                                            // parameters
-                                            memory[x + brushSize][z + brushSize] = 1; // stop it from checking any other blocks in this vertical 1x1 column.
+                                        if (!Materials.isEmpty(type)) {
```
see above comment

TheMeinerLP reviewed (changes requested):
This improves the readability for external/other developers.

Aurelien30000 committed:
Improve over(under)lay brushes
- Use boolean array instead of int array to keep track of already found blocks.
- Fix https://github.com/IntellectualSites/fastasyncvoxelsniper/issues/280

TheMeinerLP reviewed (approved):
LGTM
